### PR TITLE
8311020: Typo cleanup in Classfile API

### PR DESCRIPTION
**Pull request by liach**
This is a cleanup patch that fixes:
1. `ClassfileBuilder` should be sealed but was not; it's now sealed.
2. Various typo fixes, including spelling errors, wrong descriptions, and unnecessary full qualifications for classes in the same package.

Requesting a review from @asotona. There's a few other more complex issues discovered, which I will describe on the mailing list to elicit a discussion for best solutions.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8311020](https://bugs.openjdk.org/browse/JDK-8311020): Typo cleanup in Classfile API (**Bug** - P4)

### Reviewers
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14687/head:pull/14687` \
`$ git checkout pull/14687`

Update a local copy of the PR: \
`$ git checkout pull/14687` \
`$ git pull https://git.openjdk.org/jdk.git pull/14687/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14687`

View PR using the GUI difftool: \
`$ git pr show -t 14687`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14687.diff">https://git.openjdk.org/jdk/pull/14687.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14687#issuecomment-1610817449)

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/classfile/impl/ClassPrinterImpl.java [hunk 1]
```diff
@@ -497,7 +497,7 @@ private static Node[] elementValueToTree(AnnotationValue v) {
             case OfBoolean cv -> leafs("boolean", String.valueOf((int)cv.constantValue() != 0));
             case OfClass clv -> leafs("class", clv.className().stringValue());
             case OfEnum ev -> leafs("enum class", ev.className().stringValue(),
-                                    "contant name", ev.constantName().stringValue());
+                                    "constant name", ev.constantName().stringValue());
```
Comment (asotona) [inline src/java.base/share/classes/jdk/internal/classfile/impl/ClassPrinterImpl.java:500]: Here the same typo is repeated in ClassPrinterTest, which actually fails after the fix.

Comment (liach) [reply src/java.base/share/classes/jdk/internal/classfile/impl/ClassPrinterImpl.java:500]: Thanks for the reminder. I've checked IDE-reported typos in the tests and fixed 2; jdk.classfile tests pass on my machine now.

**Review by asotona — APPROVED** [review]
Thank you for fixing the typos.

**Pull request closed without merging**